Mathlib/Data/Polynomial/Module.lean

```diff
@@ -15,16 +15,64 @@ In this file, we define the polynomial module for an `R`-module `M`, i.e. the `R
 This is defined as a type alias `PolynomialModule R M := ℕ →₀ M`, since there might be different
 module structures on `ℕ →₀ M` of interest. See the docstring of `PolynomialModule` for details.
 
+We also define, given an element `a` in an `R`-algebra `A` and an `A`-module `M`, an `R[X]`-module
+`Module.AEval R M a`, which is a type synonym of `M` with the action of a polynomial `f`
+given by `f • m = Polynomial.aeval a f • m`. In particular `X • m = a • m`.
+
+In the special case that `A = M →ₗ[R] M` and `φ : M →ₗ[R] M`, the module `Module.AEval R M a` is
+abbreviated `Module.AEval' φ`. In this module we have `X • m = ↑φ m`.
+-/
+universe u v
+open Polynomial BigOperators
+
+namespace Module
+/--
+Suppose `a` is an element of an `R`-algebra `A` and `M` is an `A`-module.
+Then `Module.AEval R M a` is the `R[X]`-module with carrier `M`,
+where the action of `f : R[X]` is `f • m = (aeval a f) • m`.
 -/
+@[nolint unusedArguments]
+def AEval (R M : Type*) {A : Type*} [CommSemiring R] [Semiring A] [Algebra R A]
+    [AddCommMonoid M] [Module A M] [Module R M] [IsScalarTower R A M] (_ : A) := M
```

Comment:
As we discussed in #7419, I think this should be a one-field structure; semireducible type aliases are messy in Lean 4, so we should only use them where the cost is worth it.

If you do really want to keep this as not a structure, then can you add the canonical `LienarEquiv` between `M` and `AEval R M a`?

Reply:
It's seems that you are asking me to undo a change requested by another reviewer, and his request was that I undo an earlier change requested you you. I don't have the experience to make a judgement on which way is better.

Reply:
OK, it is definitely my fault. Can you please go back to the approach with a structure? Sorry for the waste of time.

Reply:
I'm happy with the compromise of sticking with the synonym and just adding the isomorphism

Reply:
Thanks for the compromise. I'll sort this out some time in the next few days (I'll be teaching all day today).

Reply:
is this ok now?

Reply:
I pushed some cleanup